app/models/provisioning_template.rb

```diff
@@ -50,11 +50,14 @@ def used_taxonomy_ids(type)
     end
   }
 
-  # TODO: review if we can improve SQL
   def self.template_ids_for(hosts)
-    hosts.with_os.map do |host|
-      host.provisioning_template.try(:id)
-    end.uniq.compact
+    oses = hosts.with_os.uniq.pluck(:operatingsystem_id)
```

Comment:
could you dry-up `hosts.with_os.uniq` expression on this and next 2 lines please?